### PR TITLE
Separate into modules for plugin and CLI to avoid versioning issues

### PR DESCRIPTION
**Pull request by IfSentient**
This PR introduces extra go submodules for the `cmd/grafana-app-sdk` and `plugin` directories, and a `go.work` to allow them to incorporate properly. It also moves the `codegen`, `kindsys`, and `internal` packages into the `cmd/grafana-app-sdk` submodule. This accomplishes two major things:
1. The CLI and library dependencies are no longer "mixed." The CLI still have some upward dependencies on the`resource` and `k8s` packages, but now using any of the libraries (`resource`, `k8s`, `operator`, etc.) does not require the dependencies of the `codegen` package or CLI. The `plugin` package has been similarly isolated because dependency versions in the root module will overwrite the ones in the submodule, and the `codegen` package requires an older version of `getkin/kin-openapi` than `plugin` requires.
2. Removes the need for the `replace` directive version pinning, which was preventing `go install` from working correctly, and _also_ required that a user of the app-sdk as a library also provided that `replace` directive in their `go.mod`.

Resolves https://github.com/grafana/grafana-app-sdk/issues/189

### CODE REVIEW

**Review by radiohead — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -29,6 +29,8 @@ $(LINTER_BINARY):
 .PHONY: test
 test:
 	go test -count=1 -cover -covermode=atomic -coverprofile=$(COVOUT) ./...
+	go test -count=1 -cover -covermode=atomic -coverprofile=$(COVOUT) ./plugin/...
+	go test -count=1 -cover -covermode=atomic -coverprofile=$(COVOUT) ./cmd/grafana-app-sdk/...
```
Comment (radiohead) [inline Makefile:33]: Good catch!

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -25,6 +25,12 @@ jobs:
     # make test
     - name: Test
       run: go test -v ./...
+    # make test
+    - name: Test CLI module
+      run: go test -v ./cmd/grafana-app-sdk/...
+    # make test
+    - name: Test plugin module
+      run: go test -v ./plugin/...
```
Comment (radiohead) [inline .github/workflows/release.yml:33]: Hm, do you think we can update our workflows to rely on `make` instead? I'm not happy with the duplication we have here.

Comment (IfSentient) [reply .github/workflows/release.yml:33]: IIRC there was some problem with using make in the workflow, but let me try and see what happens.

Comment (IfSentient) [reply .github/workflows/release.yml]: Looks like it worked, nvm, good call

**Review by radiohead — APPROVED** [review]

**Pull request merged**